tests/checkpointing/test_model_checkpoint.py

```diff
@@ -1086,7 +1067,7 @@ def __init__(self, hparams):
             super().__init__()
             self.save_hyperparameters(hparams)
 
-    model_checkpoint = ModelCheckpoint(dirpath=tmpdir, save_top_k=1, monitor="foo")
+    model_checkpoint = ModelCheckpoint(dirpath=tmpdir, save_top_k=1)
```

Comment:
The model doesn't log `foo` so this was a bug. It didn't fail before because `_is_valid_monitor_key` has `or len(metrics) == 0` and the length was 0 because we were using accessing `trainer.callback_metrics` instead of the existing `monitor_candidates`